sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CollationTypeCasts.scala

```diff
@@ -52,6 +52,11 @@ object CollationTypeCasts extends TypeCoercionRule {
       overlay.withNewChildren(collateToSingleType(Seq(overlay.input, overlay.replace))
         ++ Seq(overlay.pos, overlay.len))
 
+    case regExpReplace: RegExpReplace =>
+      val singleType = collateToSingleType(Seq(regExpReplace.subject, regExpReplace.rep))
+      val newChildren = Seq(singleType.head, regExpReplace.regexp, singleType(1), regExpReplace.pos)
```

Comment:
I find this a bit weird. Could we use `val Seq(first, third) = collateToSingleType(...)`